polaris/hub/client.py

```diff
@@ -662,18 +720,193 @@ def upload_benchmark(
+        # TODO (jstlaurent): Use the same owner for now, until the competition returns a better dataset entity
+        response["dataset"] = self._get_dataset(
+            owner, response["dataset"]["name"], ArtifactType.COMPETITION, verify_checksum=verify_checksum
+        )
+
+        if not verify_checksum:
+            response.pop("md5Sum", None)
+
+        return CompetitionSpecification(**response)
+
+    def list_competitions(self, limit: int = 100, offset: int = 0) -> list[str]:
+        """List all available competitions on the Polaris Hub.
+
+        Args:
+            limit: The maximum number of competitions to return.
+            offset: The offset from which to start returning competitions.
+
+        Returns:
+            A list of competition names in the format `owner/competition_name`.
+        """
+
+        # TODO (cwognum): What to do with pagination, i.e. limit and offset?
+        response = self._base_request_to_hub(
+            url="/v2/competition", method="GET", params={"limit": limit, "offset": offset}
+        )
+        benchmarks_list = [f"{HubOwner(**bm['owner'])}/{bm['name']}" for bm in response["data"]]
+        return benchmarks_list
+
+    def evaluate_competition(
+        self,
+        competition: CompetitionSpecification,
+        y_pred: PredictionsType,
+        result_name: str,
+        description: str = "",
+        tags: list[str] = [],
+        user_attributes: Dict[str, str] = {},
+        github_url: HttpUrlString | None = "",
+        paper_url: HttpUrlString | None = "",
+        contributors: list[HubUser] | None = [],
+        results_access: AccessType = "private",
+    ) -> CompetitionResults:
+        """Evaluate the predictions for a competition on the Polaris Hub. Target labels are fetched
+        by Polaris Hub and used only internally.
+
+        Args:
+            competition: The competition to evaluate the predictions for.
+            y_pred: The predictions for the test set, as NumPy arrays.
+                If there are multiple targets, the predictions should be wrapped in a dictionary with the target labels as keys.
+
+        Returns:
+             A `CompetitionResults` object.
+        """
+
+        y_pred = serialize_predictions(y_pred)
+        response = self._base_request_to_hub(
+            url=f"/v2/competition/{competition.owner}/{competition.name}/evaluate",
+            method="POST",
+            json={
+                "predictions": y_pred,
+                "access": results_access,
+                "polaris_version": po.__version__,
+                "name": result_name,
+                "description": description,
+                "tags": tags,
+                "user_attributes": user_attributes,
+                "github_url": github_url,
+                "paper_url": paper_url,
+                "contributors": contributors,
+            },
```

Comment:
**Comment**
I'd feel better if this was defined in a Pydantic model that was serialized to JSON for the request.

Reply:
I agree! This is essentially the results data model, right? But then just switching the `scores` attribute for the `predictions` one? Maybe we can split off a `ResultsMetadata` datamodel that is used by both the results and the object we would create here?